nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -332,12 +332,45 @@ in
 
       };
 
+      # Configuration for automatic login. Common for all DM.
+      autoLogin = mkOption {
+        type = types.submodule {
+          options = {
+            enable = mkOption {
+              type = types.bool;
+              default = cfg.displayManager.autoLogin.user != null;
+              description = ''
+                Automatically log in as <option>autoLogin.user</option>.
+              '';
+            };
+
+            user = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = ''
+                User to be used for the automatic login.
+              '';
+            };
+          };
+        };
+        
+        default = {};
```

Comment:
Thanks for merging!

One minor optimization: `autoLogin` shouldn't be an `submodule` defaulting to `{}`, it is merely a name scope, so just
```nix
autoLogin.enable = mkOption {
}
autoLogin.user = mkOption {
}
```
would be enough.

It is how it was in https://github.com/NixOS/nixpkgs/pull/81652 which I am using for ages.

Reply:
Remaining diff after resolving merge conflict there: https://github.com/NixOS/nixpkgs/pull/81652/files#diff-05435704a2787433f3120047d8dc2b35

Reply:
@volth I had to opt to use the submodule because I couldn't reference it in the manual without that type.

Reply:
Do you mean the manual generation script does not support options with `.` ?
But it is too common in nixpkgs (especially in `boot.` options), do they have problems with manual as well?
Then it is a bug to file and fix.

Reply:
It is more of addressing issue. Options have anchor assigned but arbitrary attributes on the option path do not. I usually address a group by linking to its `enable` option when available.